Build failure: programmer-calculator

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix build .#programmer-calculator` on `master`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/316661439

### Relevant log output

```console
src/main.c: In function 'main':
src/main.c:124:5: error: too many arguments to function 'init_gui'; expected 0, have 2
  124 |     init_gui(&displaywin, &inputwin);
      |     ^~~~~~~~ ~~~~~~~~~~~
In file included from src/main.c:11:
include/draw.h:29:6: note: declared here
   29 | void init_gui();
      |      ^~~~~~~~
src/main.c:127:20: error: passing argument 2 of 'signal' from incompatible pointer type [-Wincompatible-pointer-types]
  127 |     signal(SIGINT, exit_pcalc_success);
      |                    ^~~~~~~~~~~~~~~~~~
      |                    |
      |                    void (*)(void)
In file included from src/main.c:8:
/nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/signal.h:88:57: note: expected '__sighandler_t' {aka 'void (*)(int)'} but argument is of type 'void (*)(void)'
   88 | extern __sighandler_t signal (int __sig, __sighandler_t __handler)
      |                                          ~~~~~~~~~~~~~~~^~~~~~~~~
src/main.c:33:13: note: 'exit_pcalc_success' declared here
   33 | static void exit_pcalc_success();
      |             ^~~~~~~~~~~~~~~~~~
/nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/signal.h:72:16: note: '__sighandler_t' declared here
   72 | typedef void (*__sighandler_t) (int);
      |                ^~~~~~~~~~~~~~
make: *** [Makefile:41: src/main.o] Error 1
```

### Additional context

Probably due to #475479 and seems to be fixed [upstream](https://github.com/alt-romes/programmer-calculator/commit/153272c50b2491ddf25dfbfcf228a08a3b3ace69).

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.60, NixOS, 26.05 (Yarara), 26.05.20251205.f61125a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/cd6v80pj9h7r2prxx9a71y92zq88p1xx-source`

### Notify maintainers

@cjab 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pr/programmer-calculator/package.nix
+++ b/pkgs/by-name/pr/programmer-calculator/package.nix
@@ -7,20 +7,20 @@
 
 gccStdenv.mkDerivation rec {
   pname = "programmer-calculator";
-  version = "3.0";
+  version = "3.0-unstable-2025-11-06";
 
   src = fetchFromGitHub {
     owner = "alt-romes";
     repo = "programmer-calculator";
-    rev = "v${version}";
-    sha256 = "sha256-9mv8Jac6j3fKWLLCu1Bd/T5dbegUB8rRgsj9MaQhFDw=";
+    rev = "153272c50b2491ddf25dfbfcf228a08a3b3ace69";
+    sha256 = "sha256-24OYG3tVxcc/1i9HRrzW/jPY41KnKkugLziWnG1wQIw=";
   };
 
   buildInputs = [ ncurses ];
 
   installPhase = ''
     runHook preInstall
-    install -Dm 555 pcalc -t "$out/bin"
+    install -Dm 555 bin/pcalc -t "$out/bin"
     runHook postInstall
   '';
 
@@ -32,7 +32,7 @@ gccStdenv.mkDerivation rec {
       representations, sizes, and overall close to the bits
     '';
     homepage = "https://alt-romes.github.io/programmer-calculator";
-    changelog = "https://github.com/alt-romes/programmer-calculator/releases/tag/v${version}";
+    changelog = "https://github.com/alt-romes/programmer-calculator/releases/tag/v${lib.versions.majorMinor version}";
     license = lib.licenses.gpl3Only;
     maintainers = with lib.maintainers; [ cjab ];
     platforms = lib.platforms.all;

